http_outlet.go

```diff
@@ -136,13 +136,13 @@ func (h *HTTPOutlet) retryPost(batch Batch) {
 	}
 }
 
-func (h *HTTPOutlet) post(formatter HTTPFormatter, uuid string) error {
+func (h *HTTPOutlet) post(formatter HTTPFormatter) error {
 	req, err := formatter.Request()
 	if err != nil {
 		return err
 	}
 
-	req.Header.Add("X-Request-Id", uuid)
+	uuid := req.Header.Get("X-Request-Id")
```

Comment:
This is a bit leaky. I'd rather see HTTPFormatter gain the ability to return a RequestID, even if it's stored in the header down below.